os/net/lwip/src/api/sockets.c

```diff
@@ -388,6 +388,11 @@ struct lwip_sock *get_socket(int s, pid_t pid)
 	}
 
 	tcb = sched_gettcb(pid);
+	if (tcb == NULL) {
+		printf("pid: %d[%d] tcb is null\n", (int)pid, getpid());
```

Comment:
1. Let's use lwip's debug way instead of printf.
2. line 396, DEBUGASSERT(tcb) is the same as you added when CONFIG_DEBUG is enabled. If you want to add this instead of line 396, you should remove line 396 to resolve duplicated. And as line 396 does, you should check `tcb->group` as well.
3. I think you can remove `for external/webserver` from the title. This code is not for only external/webserver.